### PR TITLE
Request show overview

### PR DESCRIPTION
**Pull request by vpereira**
Convert Request#show overview to bootstrap

Overview:

screenshot_2019-01-25_16-03-17

Add Reviewer

screenshot_2019-01-28_16-52-58

Show package maintainers

screenshot_2019-01-29_14-48-21

### CODE REVIEW

**Comment by vpereira** [comment]
[Request Overview](https://trello.com/c/BDvEJdkV/1029-request-overview)

**Comment by vpereira** [comment]
> If you remove that key navigation feature, remove it for good - and also the tests for it (that are currently disabled as they flickered)

@coolo , it was still enabled for the "old ui", done now. Could you please point me where are the disabled tests related with that feature?

File: src/api/app/controllers/webui/packages/bs_requests_controller.rb [hunk 1]
```diff
@@ -9,9 +9,6 @@ def index
         requests_query = BsRequest::DataTable::FindForPackageQuery.new(@project, @package, parsed_params)
         @requests_data_table = BsRequest::DataTable::Table.new(requests_query, params[:draw])
 
-        # NOTE: This session is used by requests/show
```
Comment (DavidKang) [inline src/api/app/controllers/webui/packages/bs_requests_controller.rb]: Why did you remove this feature in the current interface?

Comment (vpereira) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: it was agreed with @coolo and @bgeuken, its not being used, not being tested and not full implemented. so its gone. @DavidKang

Comment (vpereira) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: please read the whole comments @DavidKang  https://github.com/openSUSE/open-build-service/pull/6851#pullrequestreview-196774356

Comment (DavidKang) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: @vpereira, I read this comment before, but this not explain the reason for your decision to remove this feature. BTW, I agree if it's not used we can remove it.

Comment (coolo) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: To give some background here: I implemented this feature to help me going through a lot of project requests. I also explained it to @bgeuken who extended it (with tests), but the feature still doesn't really work relying on sessions, which makes working in multiple tabs quite adventures and confusing. 

We don't know if others are using it, but we know the user story itself is there - but not too important to keep.

Comment (DavidKang) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: @coolo, thanks for the clarification.

Comment (coolo) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: And I just verified with @DimStar77 - he didn't know about this feature :)

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    = pagetitle
+    = "(#{state})"
+#description-text
+  - if description.blank?
+    %i No description set
+  - else
+    = description_wrapper(toggle_sliced_text(description))
+- if can_add_reviews
+  %p
+    %i.fas.fa-plus-circle.text-primary
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_overview.html.haml]: This should instead be:

```
= link_to('Add a review', add_reviewer_dialog_path(number: number), remote: true) do
  %i.fas.fa-plus-circle.text-primary
  Add a Review
```

So the link is properly capitalized and the font weight is normal (not bold as it is right now)

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+%h3
+  %p
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_overview.html.haml]: This can be a one-liner:

`%p #{pagetitle} (#{state})`

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+%ul.list-unstyled
+  %li
+    %i.fas.fa-user.text-success
+    Created
+    by #{user_with_realname_and_icon(request_creator, no_icon: true, short: true)} #{fuzzy_time(request_created_at)}
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: Please put the Ruby code on another line, you won't have to interpolate it in the string.

```
Created by
= user_with_realname_and_icon(request_creator, no_icon: true, short: true) 
= fuzzy_time(request_created_at)
```

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  %li
+    %i.fa.fa-info-circle.text-info{ title: 'Info' }
+    In
+    state #{link_to(request_state, { anchor: 'request_history' }, style: "color: #{request_state_color(request_state)};")}
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: Please put the Ruby code on another line, you won't have to interpolate it in the string.

We should use classes for the color instead of `style`

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    %li
+      %i.fas.fa-plus-circle.text-primary
+      Supersedes
+      - superseding.each do |supersed|
+        = link_to supersed['number'], number: supersed['number']
+  - if superseded_by
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
+      Superseded
+      by #{link_to superseded_by, number: superseded_by}
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: Please put the Ruby code on another line, you won't have to interpolate it in the string.

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    = "(#{state})"
+#description-text
+  - if description.blank?
+    %i No description set
+  - else
+    = description_wrapper(toggle_sliced_text(description))
+- if can_add_reviews
+  %p
+    %i.fas.fa-plus-circle.text-primary
+      = link_to('Add a review', add_reviewer_dialog_path(number: number), remote: true)
```
[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: It looks like it's missing.

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+%h3
+  %p
+    = pagetitle
+    = "(#{state})"
+#description-text
+  - if description.blank?
+    %i No description set
+  - else
+    = description_wrapper(toggle_sliced_text(description))
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_overview.html.haml]: We don't want to use `description_wrapper` in the webui2 views. On top of this, the `if/else` in this view is almost exactly what `description_wrapper` does.

```
  # TODO: bento_only
  def description_wrapper(description)
    if description.blank?
      content_tag(:p, id: 'description-text') do
        content_tag(:i, 'No description set')
      end
    else
      content_tag(:pre, description, id: 'description-text', class: 'plain')
    end
  end
```

We should do it directly in the view.

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        %span.fuzzy-time{ title: "#{l accept_at}" }
+        = l accept_at, format: :only_date
+        \.
+      - elsif accept_at.past?
+        This request will be automatically accepted when it enters the 'new' state.
+      - else
+        This request will be automatically accepted in #{time_ago_in_words(accept_at)}.
+  - other_open_reviews.each do |review|
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: The title doesn't bring value, let's remove it.

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      Open review for #{render partial: 'reviewer', locals: { review: review }}
+  - my_open_reviews.each do |review|
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: The title doesn't bring value, let's remove it.

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      - superseding.each do |supersed|
+        = link_to supersed['number'], number: supersed['number']
+  - if superseded_by
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
+      Superseded
+      by #{link_to superseded_by, number: superseded_by}
+  - if accept_at
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: The title doesn't bring value, let's remove it.

The icon `fa-exclamation-circle` should be paired with `text-danger`, not with `text-info`.

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+      (#{link_to('show who', '#', class: 'show_dialog', data: { target: '#maintainers_dialog' })})
+  - if superseding
+    %li
+      %i.fas.fa-plus-circle.text-primary
+      Supersedes
+      - superseding.each do |supersed|
+        = link_to supersed['number'], number: supersed['number']
+  - if superseded_by
+    %li
+      %i.fa.fa-exclamation-circle.text-info{ title: 'Alert' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: The title doesn't bring value, let's remove it.

I'm not sure though that `fa-exclamation-circle` is really the right icon here. The open review is just information here, so it should be `fa-info-circle`.

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 7 above)
[the remark this answers is not in the document]
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: please look again. I forgot to commit the partial

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 3 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: done

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 2 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: didnt work, maybe passing the block to the method link_to, you have to change something?

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml (hunk 4 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: done, in some another places too

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 8 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: I don't know who are "We" here, but, there is some javascript going on here. To use collapse do not work, because we want to show some text and then truncate it with the "+" to expand it. Collapse will break it.

Because requests use the changes as description, they are normally really long, so the way done in the project#show or package#show, IMO do not fit here. I would prefer to have this change done later,  in this Deliverable,  if we have some free cycles in the end to do it...

Comment (coolo) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: 'We' use https://getbootstrap.com/docs/4.1/components/collapse/ everywhere else

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 2 above)
Comment (bgeuken) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: Yep. In that case you can drop the first parameter, since that's essentially the block.

```ruby
= link_to(add_reviewer_dialog_path(number: number), remote: true) do
  %i.fas.fa-plus-circle.text-primary
  Add a Review
```

Comment (DavidKang) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: And also the `remote: true` is not needed anymore

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: Oh right, I forgot to remove the string `'Add a review'`. @bgeuken's version is the right one.

Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: i saw the comments now.. done. Could you please check again @dmarcoux ? Thanks anyway @bgeuken  and @DavidKang

File: src/api/app/assets/javascripts/webui2/request.js [hunk 13]
```diff
@@ -106,40 +106,9 @@ function requestShowReview() { // jshint ignore:line
...
-      case "package":
-      {
-        $('#review_user_span').hide();
-        $('#review_group_span').hide();
-        $('#review_project_span').show();
-        $('#review_package_span').show();
-      }
-        break;
-    }
+    $('span').addClass('d-none');
```
Comment (dmarcoux) [inline src/api/app/assets/javascripts/webui2/request.js]: This is going to add the class `d-none` to *all* the `span` elements. Is this really what you want to achieve?

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+- if can_add_reviews
+  %p
+    = link_to(add_reviewer_dialog_path(number: number), data: { toggle: 'modal', target: '#add-reviewer-modal' }) do
+      %i.fas.fa-plus-circle.text-primary
+      Add a review
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_overview.html.haml]: Add a **R**eview

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+    Created by
+    = user_with_realname_and_icon(request_creator, no_icon: true, short: true)
+    = fuzzy_time(request_created_at)
+  %li
+    %i.fa.fa-info-circle.text-info
+    In state
+    = link_to(request_state, { anchor: 'request_history' }, style: "color: #{request_state_color(request_state)};")
+  - unless package_maintainers.empty?
+    %li
+      %i.fas.fa-eye.text-info{ title: 'Look into' }
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: I think the title `'Look into'` doesn't bring value here. I would remove it.

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+      = pluralize(package_maintainers.size, 'package maintainer')
+      (#{link_to('show who', '#', class: 'show_dialog', data: { toggle: 'modal', target: '#show-maintainers-modal' })})
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: The link `(show who)` is a bit weird. I would do this instead:

 ```
= link_to('#', class: 'show_dialog', data: { toggle: 'modal', target: '#show-maintainers-modal' }) do
  = pluralize(package_maintainers.size, 'package maintainer')
```

It would look like this
suggestion

File: src/api/app/assets/javascripts/webui2/request.js (hunk 13 above)
Comment (vpereira) [reply src/api/app/assets/javascripts/webui2/request.js]: nope, i narrowed it down to ```#add-reviewer-modal span```

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml (hunk 14 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_overview.html.haml]: done

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml (hunk 15 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: look, eye, get it? at the obs-pattern we have the magnifying glass for show. The eye was already there and I do not have a strong opinion against it.. any idea @dmarcoux

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml (hunk 16 above)
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: done

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+  - if superseding
+    %li
+      %i.fas.fa-plus-circle.text-primary
+      Supersedes
+      - superseding.each do |supersed|
+        = link_to supersed['number'], number: supersed['number']
+  - if superseded_by
+    %li
+      %i.fa.fa-exclamation-circle.text-info
```
[the remark this answers is not in the document]
Comment (vpereira) [reply src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: no strong opinion here, but as the original is using ```text-danger``` I changed it here too.. done!

File: src/api/app/views/webui2/webui/request/_show_side_links.html.haml (hunk 15 above)
Comment (dmarcoux) [reply src/api/app/views/webui2/webui/request/_show_side_links.html.haml]: Why not the `fa-info-circle`?

File: src/api/app/controllers/webui/packages/bs_requests_controller.rb (hunk 1 above)
Comment (Ana06) [reply src/api/app/controllers/webui/packages/bs_requests_controller.rb]: Please add the reason why this is removed in the commit message. It is important information you can not deduce from the code. In general a description in the commit messages is a good idea :wink: Remember to include **why** the changes are needed :smile:

File: src/api/app/views/webui2/webui/request/_show_package_maintainers.html.haml [hunk 18]
```diff
@@ -0,0 +1,13 @@
+.modal.fade#show-maintainers-modal{ tabindex: -1, role: 'dialog', aria: { labelledby: 'confirm-modal-label', hidden: true } }
+  .modal-dialog.modal-dialog-centered{ role: 'document' }
+    .modal-content
+      .modal-header
+        %h5.modal-title#confirm-modal-label
+          Package Maintainers
+      .modal-body
+        - package_maintainers.each do |package_maintainer|
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/request/_show_package_maintainers.html.haml]: Nitpick, but I would prefer

```ruby
- package_maintainers.pluck(:login).each do |package_maintainer|
  %p
    = package_maintainer
```

Or `package_maintainers.select(:login).each` ...

**Comment by vpereira** [comment]
@dmarcoux  and @bgeuken  please look into again

File: src/api/app/views/webui2/webui/request/_show_overview.html.haml [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+
+#description-text
+  - if description.blank?
+    %i No description set
+  - else
+    = description_wrapper(toggle_sliced_text(description))
+- if can_add_reviews
+  %p
+    %i.fas.fa-plus-circle.text-primary
+      = link_to('Add a review', add_reviewer_dialog_path(number: number), data: { toggle: 'modal', target: '#add-reviewer-modal' })
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/request/_show_overview.html.haml]: The font of the link text looks different to what we usually use. I think this changes if you swap around icon and link, like this:

```ruby
= link_to(add_reviewer_dialog_path(number: number), data: { toggle: 'modal', target: '#add-reviewer-modal' }) do
  %i.fas.fa-plus-circle.text-primary
    Add a review
```

**Comment by vpereira** [comment]
I set it to WIP, since after some javascript refactoring we broke the "add review" in the old UI. I will do it later.
